src/MessageRequest.jl

```diff
@@ -51,15 +51,16 @@ function request(::Type{MessageLayer{Next}},
     return request(Next, url, req, body; iofunction=iofunction, kw...)
 end
 
-const USER_AGENT = Ref{String}()
+const USER_AGENT = Ref{Union{String, Nothing}}("HTTP.jl/$VERSION")
```

Comment:
Should this not be the version of `HTTP.jl` instead of the Julia version?

Reply:
Why?

Reply:
I think that setting this to the `HTTP.jl` version rather than the Julia version makes more sense. If a user ever has an issue down the line, it makes more sense to tie the default user-agent to the package rather than the version of Julia.

Tying the default user-agent to the package also seems to be more in-line with the spirit of what the identification represents. Although I can understand the argument 

> https://en.wikipedia.org/wiki/User_agent#User_agent_identification

Reply:
`curl`, `wget` and `python-requests` uses its own version rather than the compiler too.